cmake/FindGTS.cmake

```
@@ -65,7 +65,19 @@ else()
   endif()
 
   # 2.1 Need glib library
+  find_path(GLIB_INCLUDE_DIR
+    NAMES glib.h
+    PATH_SUFFIXES glib-2.0)
+  find_path(GLIBCONFIG_INCLUDE_DIR
+    NAMES glibconfig.h
+    PATH_SUFFIXES lib/glib-2.0/include)
   find_library(GLIB_LIBRARY glib-2.0)
+  if (GLIB_INCLUDE_DIR)
+    list(APPEND GTS_INCLUDE_DIRS "${GLIB_INCLUDE_DIR}")
+  endif()
+  if (GLIBCONFIG_INCLUDE_DIR)
+    list(APPEND GTS_INCLUDE_DIRS "${GLIBCONFIG_INCLUDE_DIR}")
+  endif()
   list(APPEND GTS_LIBRARIES "${GLIB_LIBRARY}")
```

```suggestion
  if (GLIB_INCLUDE_DIR)
    list(APPEND GTS_INCLUDE_DIRS "${GLIB_INCLUDE_DIR}")
  endif()
  if (GLIBCONFIG_INCLUDE_DIR)
    list(APPEND GTS_INCLUDE_DIRS "${GLIBCONFIG_INCLUDE_DIR}")
  endif()
  find_library(GLIB_LIBRARY glib-2.0)
  list(APPEND GTS_LIBRARIES "${GLIB_LIBRARY}")
```

Minor comment, but I think it'd be better to append include directories before searching for and including the library. That way, it doesn't appear like the library needs to be searched for first before adding the include directories.

---

Also, should we do a similar check with the result of the `find_library` call in case that isn't found?

---

I also wonder if it would be good to display some sort of warning message if the include directories or library can't be found. That way, we can have some debugging information for something like CI.

---

> better to append include directories before searching for and including the library.

I did the other way around, finding the library before appending the includes, because I think the `find_library` call is the one that initially populates `GTS_INCLUDE_DIRS`. See 2d90059.

> should we do a similar check with the result of the find_library call in case that isn't found?

Done on 2d90059

>  I also wonder if it would be good to display some sort of warning message if the include directories or library can't be found.

Honestly I don't know if not finding these directories is a mistake. Our Windows CI has been passing without it. So I think it may be needed by some setups, like with `conda-forge`, but not by others, like with `vcpkg`. If not finding the directories is ok, then a warning could be misleading.